### ISSUE TITLE
As a graduate student, I am required to cite my sources using page numbers.

### ISSUE THREAD

**Issue by sueneu** [w3c/publishingcg#51]
## Introduction

As a graduate student, I am required to cite my sources and include page numbers.  

*Explanation:*
Reflowable books without page numbers are frustrating for people who must follow standards like those of the IEEE or APA.  

## Detail
1. **Audience** Graduate students required the use standardized citation styles for research papers.<br/> Researchers whose work requires citations that include page numbers.<br/> Librarians and media specialists who maintain collections and recommend books for research.<br/><br/>

2. **Use Case** Clinical Psychology graduate students who must write papers that adhere to the APA citation standard. Excerpt:<br/>

> SHORT QUOTATIONS
If you are directly quoting from a work, you will need to include the author, year of publication, and page number for the reference<br/> I[n-Text Citations: The Basics, Purdue University ](https://owl.purdue.edu/owl/research_and_citation/apa_style/apa_formatting_and_style_guide/in_text_citations_the_basics.html)<br/><br/>

3. **Benefits** If more publication formats conform to existing citation requirements, there would be a wider range of materials available to me as a researcher. <br/>
Page numbers in indexes provide important information about book content and relevancy.*<br/>
A reflowable, customizable format with page numbers could satisfy scholarly citation requirements and be accessible.<br/>
Publishers could fulfill legal obligations of accessibility while giving the higher education market citable publications.<br/>
Librarians and Media Specialists who maintain collections for researchers would have more options for acquiring publications if more formats had page numbers.<br/><br/>
<hr>
* For example researchers may decide a publication isn't relevant if the term they are searching for appears only on pages numbered with roman numerals. This means the term they are searching for is only in the frontmatter and used to give overall context.

**Comment by gautierchomel** [comment]
I see two uses cases here: 

* if the print version exist, it's up to the publisher to provide the information. It happens often and in such cases I have to buy EPUB + PDF or print, I read the EPUB and refer to printed (pdf) version to catch page numbers. Gladly I can read prints and pdfs! Also I can pay the extra cost...
* if no print version exist, it's an academic practices concern and citation standards might consider evolving. It also applies to citation of webpages. I can't find a reference but remember of academics doing reference to paragraphs instead (example: _third paragraph of chapter 3_).

**Comment by sueneu** [comment]
> if the print version exist, it's up to the publisher to provide the information. It happens often and in such cases I have to buy EPUB + PDF or print, I read the EPUB and refer to printed (pdf) version to catch page numbers. Gladly I can read prints and pdfs! Also I can pay the extra cost...
True! I wonder if the cost of buying access to both ebooks and PDF/print versions would be high for an institution. 

> if no print version exist, it's an academic practices concern and citation standards might consider evolving. It also applies to citation of webpages. I can't find a reference but remember of academics doing reference to paragraphs instead (example: third paragraph of chapter 3).

I imagine citation standards will evolve. We have no control over that. In the meantime, if we offered a page-numbering option in web-delivered materials we can accommodate students and researchers until citation standards catch up to newer media.

We could consider “screens” and “pages” as separate containers. At a large font size, one may need to view 3 screens of text to read a complete “page” as defined by the author. “Page” becomes a subset of “chapter” and not representative of a physical piece of paper.

Gautier, do you know if there are similar page number citation requirements for French/EU researchers?

Susan

**Comment by mickael-menu** [comment]
Doesn't [the `page-list` navigation document](http://kb.daisy.org/publishing/docs/navigation/pagelist.html) already fit this use case? Though most EPUB publications don't have a page list and many reading systems ignore them.

**Comment by northgardner** [comment]
Just want to add another closely related user story here. "As a student, instructor, or discussion group participant, I need to be able to identify locations in a book for a group or class where some participants have a paper/PDF version and some have an EPUB." In this case, the user is citing a very specific location in an informal way, but may require even more precision than some formal citation styles, so that other participants do not have to scan through pages worth of text to find the relevant passage or figure.

**Comment by sueneu** [comment]
@mickael-menu if the page numbers from the `page-list` were visible while reading a resource, they might fit the bill. 

And could those be differentiated from the "page numbers" reading systems assign to screens, which change with user preferences and can't be used for citation?

**Comment by mickael-menu** [comment]
> And could those be differentiated from the "page numbers" reading systems assign to screens, which change with user preferences and can't be used for citation?

Absolutely, they are unrelated. You can try it out already in [Thorium](https://www.edrlab.org/software/thorium-reader/) with this book which contains a `page-list`: 
[childrens-literature.epub.zip](https://github.com/w3c/publishingcg/files/10247045/childrens-literature.epub.zip). There's a way to jump to existing print pages, as shown in this video:

https://user-images.githubusercontent.com/58686775/208134291-479e940a-420e-4f0e-9a06-7cd80a6add1b.mov

> if the page numbers from the page-list were visible while reading a resource, they might fit the bill.

In the above book, the page number labels are actually part of the book, Thorium doesn't do anything. However for other books, displaying an annotation could definitely be handled by reading systems. I'll take a look in Readium to see if we can improve this.

**Comment by mickael-menu** [comment]
I experimented with displaying the page numbers in the Android Readium toolkit, I'll add this in our example app. But reading apps using Readium will need to opt-in to display the page numbers and publishers need to provide books with `page-list` for this to be useful, of course.

![Screenshot_1671214528](https://user-images.githubusercontent.com/58686775/208163015-f6debd4a-9cfc-4b42-b7c6-dfd3d88ac837.png)

**Comment by sueneu** [comment]
@mickael-menu This looks very promising! Are the page numbers are at the beginning of the page or at the end?

**Comment by mickael-menu** [comment]
@sueneu It depends how the book is authored, but I would expect that it matches the beginning of a page, as you can see with the first page "169" in the screenshot.

**Comment by spudthebud** [comment]
I recommend approaching this design task from the perspective of print-disabled readers, for example a blind screen reader user, a partially sighted reader who magnifies the screen, or print-disabled readers who can see but can't use a mouse.   

In higher education, getting page number is very important for citation purposes.  Some people dismiss the need for page numbers, but those who can't get page numbers in higher education are at a serious disadvantage. 

Blind Canadian graduate students have told me about the problems of getting page numbers in university. A recent undergraduate student, who is also blind, also confirmed.  

At a national Canadian library conference, a blind grad student explained that the inability to get page numbers was a serious problem for blind readers in university, like [here](https://www.youtube.com/watch?v=0uC-arfXCmQ&t=109s)  and [here](https://youtu.be/0uC-arfXCmQ?t=151).  

At another Canadian library conference, a panel of blind scholars at all stages of academic careers described journal article reading experiences. A faculty member described getting page numbers as "[a monumental problem](https://vimeo.com/770786119)" (at 12:10).  

### Solution
A technique that would address these needs is for reading systems to give readers options about displaying page numbering. One option could be to inject the page numbers directly between words when they want page numbers or remove them when they don't. I have seen this injection technique in some places in academic publishing.  

### Example of in-text page numbers
Here is an example on Project Muse:

https://user-images.githubusercontent.com/8799391/208209784-318ff152-1a7f-437f-9e75-2377b5ddc53e.mp4

**Comment by dalerrogers** [comment]
Page numbers would be dependent on the viewport size. I’m taking a class now online. The assignments refer to page numbers in the printed text. That threw off students that are using eBooks. Pagination will differ from device to device to device.

Best Regards,

Dale Rogers, M.Ed., CIW
Designer
eLearning Developer
***@***.***
http://dalerogers.me/
https://www.linkedin.com/in/dalerrogers/

From my iPhone. Pardon my thumbs.

________________________________

**Comment by spudthebud** [comment]
Epub can have page breaks in the text that correspond to print page breaks.  They aren't displayed visually. 

Here's an example of that from **Over Our Dead Bodies: Port Arthur and Australia's Fight for Gun Control** by Simon Chapman, Sydney University Press
### 
Epub
`
It irrevocably changed the way the great majority of the community saw <span class="pagebreak-rw" id="pagebreak-rw_19" epub:type="pagebreak"  role="doc-pagebreak">19</span>the gun lobby. These men`
### 
Print book
<img width="621" alt="image" src="https://user-images.githubusercontent.com/8799391/208232247-3c4828f2-ffb5-49b7-b9d3-fdc2b2fa429e.png">

If a reading system could allow the user to toggle on or off the page numbers in the text, than that reading system would remove a disadvantage for print-disabled readers in higher education, like those I cited earlier.

**Comment by spudthebud** [comment]
Here's a video documenting a toggle technique for adding page numbers into HTML-based versions when there is a corresponding print version with page numbers.  

The reader is given the option to turn on or off page numbers.  When on, the page numbers are displayed in the text.
This technique would work when webpages are magnified at high levels.

In the video, I mention that  blind members of universities have spoken about the problem of getting page number problems at Canadian library conferences. Here are direct links.  Ashley Shaw, a masters student at Wilfrid Laurier University, raises the problem of page numbers [at 1:49](https://www.youtube.com/watch?v=0uC-arfXCmQ&t=109s) and [2:31](https://youtu.be/0uC-arfXCmQ?t=151).  And Dr. Cynthia Bruce of Concordia University describes the issue of page numbers as "[monumental](https://vimeo.com/770786119#t=12m9s)" at 12:09.

This technique could address these disadvantages.

### Toggle Technique: Demonstration
I've documented this technique with a screen reader in the following video, which I've described for those accessing it non-visually.  

https://user-images.githubusercontent.com/8799391/208252971-a7f46c03-a0ba-4d97-8854-241547b5d36a.mp4

### Transcript:
This video documents a toggle technique for providing page numbers in eBooks in a manner that corresponds to the page numbers in their print counter part.

The inability to get page numbers from eBooks or eJounrals is a major problem for print-disabled readers in university settings. 

This has been stated by blind students and faculty presenting at Canadian library conferences. There are recordings of these presentations online.

As context, I am sighted and JAWS certified.
I will describe this video for those accessing it non-visually.

I have the Chrome browser open. We are on the Sage Knowledge platform at an eBook, titled the Encyclopedia of Law and Higher Education.
We are at the entry for Affirmative Action.
Also on the screen is a smaller window showing my hands on the keyboard. This is to acknowledge that not everyone interacts with computers using a mouse.

I am now opening up another window that shows a view of the physical version of the same book showing the same entry.
This preview of the hardcover copy is provided through Amazon
We can see the entry spans page 12 and 13.
We can see the page break happens as follows: 

In higher education admissions cases, strict scrutiny, the highest standard of review used by the courts, is used. Start page 13. Under strict scrutiny, and it goes on from there.

I've closed the Amazon window, I've open a smaller window with a visual reminder of where page 13 happens.

And we are back at the entry on the Sage Knowledge platform

As I scroll through the entry's web page, we can see there are no page numbers.
However, we can see that there is a link on the page which says "Show page numbers"
I will now turn on the voice of JAWS and activate that.
[Stepping through the text with with JAWS.]
JAWS: "... standard of review used by the courts, is used. Left bracket page 13 right bracket. Under strict scrutiny..."

### References

Bruce, C., Marsden, M., Martiniello, N., Schiafone, C., Shaw, A., & Weiler, M. (October 3, 2022).  [Ensuring that the “World’s Knowledge is Accessible By All”: Canadian Blind Scholars Share their Experiences of Journal and other Digital Content](https://vimeo.com/770786119)
. Panel presentation to the Canadian Research Knowledge Network.

Shaw, A., Weiler, M., Kuwala, B., Thomas, M., & Kwak, A. (October 13, 2022). [PDF Pitstops in Libraries: An Essential Feature for Blind Users](https://www.youtube.com/watch?v=0uC-arfXCmQ). Access Conference, Carleton University.

**Comment by sueneu** [comment]
@dalerrogers 
Exactly right! It looks like displaying print page numbers would make referencing text for assignments easier for online learners like you. Would a solution like @mickael-menu's example be helpful?

> Page numbers would be dependent on the viewport size. I’m taking a class now online. The assignments refer to page numbers in the printed text. That threw off students that are using eBooks. Pagination will differ from device to device to device.

**Comment by dalerrogers** [comment]
I think the easiest solution is for educators to recognize the limitations, and variety, of reference content. Instead of requiring students to read pages 146-200. Perhaps direct them to chapter 9 (for example), starting at the paragraph beginning with “some text”. Then, if the ebook has chapter titles set up, they can do a search for the block of content easily.

I think it’s more of a Universal Design approach  for instructional designers than requiring a technology to do something it can’t. If page numbers are arbitrary based on viewport (and font size settings), then don’t use page numbers as a reference. The page text will be constant, the page numbers will not.


Best Regards,

Dale Rogers, M.Ed., CIW
Designer
eLearning Developer
***@***.***
http://dalerogers.me/
https://www.linkedin.com/in/dalerrogers/

From my iPhone. Pardon my thumbs.

________________________________

**Comment by spudthebud** [comment]
@dalerrogers page numbers can easily be in e-content if there is a print equivalent. They are put in the underlying HTML code so don't depend on viewport. This is demonstrated in the video in [comment](https://github.com/w3c/publishingcg/issues/51#issuecomment-1356353856).  

Universal design would try to proactively build something into the environment that removes a disadvantage affecting a disabled individual.   

For academic, research, or scholarly organizations concerned about resources, it is a better to procure a reading platform that gives their readers the option to control how they get page numbers rather than investing in outreach to convince scientists, researchers, scholars, educators, to change a convention of their academic reading and writing.

I predict this market of readers will more quickly adopt a reading platform that has the option to display page numbers than platforms that require them to change their conventions of reading and writing.  

I also predict blind and print disabled readers will voice fewer concerns with organizations, such as universities, that adopt reading platforms that removes disadvantages for them.

**Comment by sueneu** [comment]
@spudthebud @dalerrogers Building in the ability for writers/publishers to include persistent page numbers would match most people's expectations of a book. This would give publishers the option to adapt their products to the user's existing mental model.

> [Mental models are one of the most important concepts in human–computer interaction](https://www.nngroup.com/articles/mental-models/) (HCI). -Jacob Nielsen, nngroup

Perhaps someday mental models will change and new modes of citation and location will emerge. For now, the option to implement persistent page numbers can make reflowable books more useful for higher ed.

**Comment by gautierchomel** [comment]
> Gautier, do you know if there are similar page number citation requirements for French/EU researchers?

@sueneu yes, it affects mainly humanities and social sciences but with some flexibility. The main problem is still with psychology curriculum requiring APA. But my understanding is that APA is evolving regularly. Also  APA page number requisite applies to printed books, not digital contents. I can use APA Style to cite webpages and in that case page number does not apply, so actually using APA with digital contents without a page number is possible. 

In fact here in France mains online reading portals like cairn.info and openedition.org don't provide page number for online versions of printed books. They provide a range of page for an article (follow the link to see an [exemple page 17-35](https://books.openedition.org/pup/38380)) and display paragraph numbers. 

I still advocate for a division of this problematic in two use cases:

* books that have a printed version previous to a digital one should include print page numbers as the EPUB specification allows it. A publisher not doing so is responsible for the non usability of the book in academic (or scholars) context. This is to be addressed thru information, communication and training. Providing visual rendering of print page number as showed by [Mickael experiment](https://github.com/w3c/publishingcg/issues/51#issuecomment-1355358568) could be a plus for reading system. Personally I prefer not to be distracted by print page numbers and look for them when needed (thru _Where Am I_ or _Go to page_ functions).

* books that don't have a prior printed version should not try to reproduce a print citation experience as it would make print page number impossible to understand. Long term strong solution here might be [EPUB Canonical Fragment Identifiers](https://w3c.github.io/epub-specs/epub33/epubcfi/), but for short term I would use book structure (chapter, sub chapter and paragraphs) and add a note that this book has no print page number. 

Actually reasons for not adding print page numbers are cost relatives. I fear that allowing an automated (not print) false page numbering would encourage to apply that to any book and we'll loose the print page reference witch is crucial in a mixed environment where some of the students use print versions and some others use digital version.

**Comment by gautierchomel** [comment]
> Perhaps someday mental models will change and new modes of citation and location will emerge. For now, the option to implement persistent page numbers can make reflowable books more useful for higher ed.

I don't believe in an average mental model and I see many persons in difficulties because of an imposed mental model supposed to be the most common. 

> [The End of Average: How We Succeed in a World That Values Sameness](http://www.toddrose.com/books#/endofaverage/). Todd Rose, Jan 2016 · HarperCollins

**Comment by sueneu** [comment]
@gautierchomel 

> I don't believe in an average mental model and I see many persons in difficulties because of an imposed mental model supposed to be the most common.

Adding the technical capability for page numbers isn't the same thing as imposing page numbers. It would allow publishers that option if their market desires them. I do not believe that device-independent page numbers will stifle creativity or promote intellectual conformity.

> books that have a printed version previous to a digital one should include print page numbers as the EPUB specification allows it. A publisher not doing so is responsible for the non usability of the book in academic (or scholars) context. This is to be addressed thru information, communication and training. 

I agree that the ultimate responsibility for usability lies with the reading system. Does the WG or TF provide information and training? That may be a better response than changing the ePub Spec. 

> books that don't have a prior printed version should not try to reproduce a print citation experience as it would make print page number impossible to understand. Long term strong solution here might be [EPUB Canonical Fragment Identifiers](https://w3c.github.io/epub-specs/epub33/epubcfi/), but for short term I would use book structure (chapter, sub chapter and paragraphs) and add a note that this book has no print page number.

Fragment identifiers make sense for books that are "born digital."  The APA and IEEE will probably make changes in the future to their citation requirements. We— and current researchers and scholars have no control over that. 

Are we providing publishers with the tools to meet their readers' needs?

**Comment by spudthebud** [comment]
@gautierchomel just because OpenEditions.org or cairn.info are using a technique in public doesn't mean the technique removes  disadvantages or are even meets accessibility guidelines. 

For example, when magnifying OpenEditions.org to 150% and Cairn.Info to 200% in Chrome the paragraph numbers are removed.  How does that not disadvantage low vision readers?  And I tried both sites with Safari and VoiceOver, and every paragraph is now spoken with a number.  Sighted people may be able to ignore the paragraph numbers but are screen reader users now required to hear every paragraph? What would the reading experience be like for a 300 page book where every paragraph number is spoken?  And do the paragraphs numbers affect the reading flow of readers with dyslexia?

What's the reason for not giving users of reading platforms a variety of choices? If page numbers are available in a print form (e.g., PDF, paper), then the user can toggle on or off page numbers in the e-version.  If page numbers aren't available, then the user can toggle on or off paragraph numbers.  

But in all design scenarios, I think it's important to focus on providing features that remove disadvantages for print-disabled readers, and verifying it.

**Comment by sueneu** [comment]
@gautierchomel @spudthebud You make a good point about screen readers and page numbers. That sounds awful. 

What if print/persistent page numbers appear only when a reader interacts with the text? For instance when making a `note` or `annotation`. I've noticed that text can be appended on `copy ` like this from Apple Books—

> “78The electric alarm clock sat where Winna couldn’t see it,”
> 
> Excerpt From
> Speculation
> Nisi Shawl
> This material may be protected by copyright.

Perhaps similar tech could add the print page numbers needed for scholarly citation on certain interactions.

**Comment by gautierchomel** [comment]
@sueneu That how it works actually. 

Thorium does not yet has annotations implemented but you'll get print page number using the _Where Am I_ function and you can reach one page thru _Go to page_ menu. Screen readers will announce "separator, 3" when starting page 3. You can see the [Thorium documentation](https://edrlab.github.io/thorium-reader-doc/210_reading/212_whereAmI/) for screenshots and details. 

We hope to get funding for annotation implementation and clearly page number will be part of the export. I also hope we'll implement visual display of page numbers.

I feel that this print page feature is little known and often misunderstood, but I'm certain we should be careful and keep it well separated of any other persistent pagination. 

Also I fear that proposing other system to add a persistent pagination will have as effect that print page reference will no more be implemented because it cost much more than adding automated pagination each xxx words. As a user I will be affected because I'll lost the ability to cross reference between print and digital versions and I will need both versions depending on the instruction or citation made. That means a blind student will lost access to a born accessible version of the printed book.

**Comment by sueneu** [comment]
@gautierchomel I love where Thorium is going

- The print page number feature already exists. It could be more widely used and should be thoughtfully implemented.
> I feel that this print page feature is little known and often misunderstood, but I'm certain we should be careful and keep it well separated of any other persistent pagination.

@TzviyaSiegman What is the role of EdTF/PublishingWG? 

- Does EdTF's work end when we make sure the Ebook spec contains features that users want? 
If so, then we may need to do nothing further. We would leave the UI and implementation to reading systems. 

- Do we reach out to publishers, reading systems, and authors to determine why the feature isn't used? Perhaps the spec needs some adjustments that we then recommend to the WG.

- Do we offer guidance/education for Ebook spec features? We could develop and propose an outreach plan.

**Comment by wareid** [comment]
The EPUB WG's Virtual Locators Taskforce spent a little bit of time trying to figure out the question of page numbering (locators) in digital books. We have a potential solution and algorithm outlined [in this document](https://docs.google.com/document/d/11GypOjE9xOTaINATl5bxVIA3Mc9jzNBGCr6GT_KNaQ4/edit?usp=sharing). That document also contains use cases, and all of our (disorganized) thoughts on page numbering in publications. 

@sueneu I think your questions are probably more appropriate for the mailing list than this issue, otherwise we'll get off topic :).
